File: jpa/Data-JPA/data-jpa/src/main/java/study/datajpa/entity/Team.java
package study.datajpa.entity;

import javax.persistence.*;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter @Setter
@NoArgsConstructor
public class Team {
    @Id
    @GeneratedValue
    @Column(name="team_id")
    private Long id;
    private String name;

    @OneToMany(mappedBy = "team")
    private List<Member> members = new ArrayList<>();

    public Team(String name){
        this.name = name;
    }
}

File: jpa/jpa/src/main/kotlin/practice/jpa/domain/Team.kt
package practice.jpa.domain

import org.hibernate.annotations.BatchSize
import javax.persistence.*

@Entity
@Table(name = "Team")
class Team {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id : Long ? = null

    @Column(name = "team_name")
    var name : String ? = null


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id")
    var organization : Organization ? = null

    //@BatchSize(size = 200)
    @OneToMany(mappedBy = "team", fetch = FetchType.LAZY)
    var members : MutableList<Member> = mutableListOf()

}